plsync/mlabconfig.py

```diff
@@ -45,13 +64,152 @@ def parse_flags():
+
+def write_a_record(output, hostname, ipv4):
+    output.write(format_a_record(hostname, ipv4))
+    output.write('\n')
+
+
+def format_a_record(hostname, ipv4):
+    return '%-32s  IN  A   \t%s' % (hostname, ipv4)
+
+
+def write_aaaa_record(output, hostname, ipv6):
+    output.write(format_aaaa_record(hostname, ipv6))
+    output.write('\n')
+
+
+def format_aaaa_record(hostname, ipv6):
+    return '%-32s  IN  AAAA\t%s' % (hostname, ipv6)
+
+
+def export_router_and_switch_records(output, sites):
+  comment(output, 'router and switch v4 records.')
+  for i, site in enumerate(sites):
+      write_a_record(output, 'r1.' + site['name'], site.ipv4(index=1))
+      write_a_record(output, 's1.' + site['name'], site.ipv4(index=2))
+
+
+def export_pcu_records(output, sites):
+  comment(output, 'pcus v4')
+  for site in sites:
+      # TODO: change site['nodes'] to a pre-sorted list type.
+      for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+          write_a_record(output, node['pcu'].recordname(), node['pcu'].ipv4())
+
+
+def export_server_records(output, sites):
+  export_server_records_v4(output, sites)
+  export_server_records_v4(output, sites, decoration='v4')
+  export_server_records_v6(output, sites)
+  export_server_records_v6(output, sites, decoration='v6')
+
+
+def export_server_records_v4(output, sites, decoration=''):
+  comment(output, 'hosts v4%s' % (' decorated' if decoration else ''))
+  for site in sites:
+      # TODO: change site['nodes'] to a pre-sorted list type.
+      for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+          write_a_record(output, node.recordname(decoration), node.ipv4())
+
+
+def export_server_records_v6(output, sites, decoration=''):
+  comment(output, 'hosts v6%s' % (' decorated' if decoration else ''))
+  for site in sites:
+      # TODO: change site['nodes'] to a pre-sorted list type.
+      for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+          if node.ipv6_is_enabled():
+              write_aaaa_record(
+                  output, node.recordname(decoration), node.ipv6())
+
+
+def export_experiment_records(output, sites, experiments):
+  for experiment in experiments:
+    if experiment['index'] is None:
+      # Ignore experiments without an IP address.
+      continue
+    export_experiment_records_v4(output, sites, experiment)
+    export_experiment_records_v4(output, sites, experiment, decoration='v4')
+
+    export_experiment_records_v6(output, sites, experiment)
+    export_experiment_records_v6(output, sites, experiment, decoration='v6')
+
+
+def export_experiment_records_v4(output, sites, experiment, decoration=''):
+  comment(output, '%s v4%s' % (experiment.dnsname(), (
+      ' decorated' if decoration else '')))
+  for site in sites:
+    # TODO: change site['nodes'] to a pre-sorted list type.
+    for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+        # TODO: should sitenames exclude mlab4's?
```

Comment:
Oh, the site names are those records like this (which exclude the machine names):
```
$ host ndt.iupui.lga04.measurement-lab.org
ndt.iupui.lga04.measurement-lab.org has address 173.205.4.75
ndt.iupui.lga04.measurement-lab.org has address 173.205.4.88
ndt.iupui.lga04.measurement-lab.org has address 173.205.4.101
ndt.iupui.lga04.measurement-lab.org has address 173.205.4.114
```

Since mlab4's are included in this list, I was wondering in that TODO whether the site names have much production value.